webpack/ForemanInventoryUpload/Components/PageHeader/components/ToolbarButtons/__tests__/ToolbarButtons.test.js

```diff
@@ -1,60 +1,78 @@
 import React from 'react';
-import { Provider } from 'react-redux';
-import configureMockStore from 'redux-mock-store';
-import thunk from 'redux-thunk';
-import { screen, render } from '@testing-library/react';
+import { screen } from '@testing-library/react';
+import { rtlHelpers } from 'foremanReact/common/rtlTestHelpers';
 import ToolbarButtons from '../ToolbarButtons';
+import { useAdvisorEngineConfig } from '../../../../../../common/Hooks/ConfigHooks';
+import { selectSubscriptionConnectionEnabled } from '../../../../InventorySettings/InventorySettingsSelectors';
 
-const middlewares = [thunk];
-const mockStore = configureMockStore(middlewares);
-
-const renderOptions = {
-  API: {
-    INVENTORY_SETTINGS: {
-      response: { subscriptionConnectionEnabled: true },
-    },
-  },
-  ForemanRhCloud: {
-    inventoryUpload: {
-      accountsList: {
-        CloudConnectorStatus: {},
-      },
-    },
-  },
-};
+// Mock the config hook
+jest.mock('../../../../../../common/Hooks/ConfigHooks', () => ({
+  useAdvisorEngineConfig: jest.fn(),
+}));
+
+// Mock the selector
+jest.mock('../../../../InventorySettings/InventorySettingsSelectors', () => ({
+  selectSubscriptionConnectionEnabled: jest.fn(),
+}));
+
+// Mock child components to isolate ToolbarButtons testing
+jest.mock(
+  '../../SyncButton',
+  () =>
+    function MockSyncButton() {
+      return <div data-testid="sync-button">Sync all inventory status</div>;
+    }
+);
+jest.mock(
+  '../../CloudConnectorButton',
+  () =>
+    function MockCloudConnectorButton() {
+      return (
+        <div data-testid="cloud-connector-button">
+          Configure cloud connector
+        </div>
+      );
+    }
+);
+
+const { renderWithStore } = rtlHelpers;
 
 describe('ToolbarButtons', () => {
-  test('when subscription connection is enabled', () => {
-    const store = mockStore(renderOptions);
-
-    render(
-      <Provider store={store}>
-        <ToolbarButtons />
-      </Provider>
-    );
-    expect(screen.queryAllByText('Configure cloud connector')).toHaveLength(1);
-    expect(screen.queryAllByText('Sync all inventory status')).toHaveLength(1);
+  test('renders both buttons when subscription connection is enabled and not in IOP mode', () => {
+    useAdvisorEngineConfig.mockReturnValue(false);
+    selectSubscriptionConnectionEnabled.mockReturnValue(true);
```

Comment:
The same comment: you can add the setting to the store rendering function, and then you won't have to mock the selectors.
Mocking the selectors means you never test them properly, and I don't see a good reason to do it, unless the selector is really complicated.